avaframe/com1DFA/DFAfunctionsCython.pyx

```diff
@@ -741,32 +769,56 @@ def updatePositionC(cfg, particles, dem, force, DT):
   log.debug('Entrained DFA mass: %s kg', np.asarray(massEntrained))
   particles['kineticEne'] = TotkinEneNew
   particles['potentialEne'] = TotpotEneNew
-  if peakKinEne < TotkinEneNew:
-    particles['peakKinEne'] = TotkinEneNew
-  if particles['peakMassFlowing'] < massFlowing:
-    particles['peakMassFlowing'] = massFlowing
-
-  if cfg['stopCritType'] == 'massFlowing':
-    value = massFlowing
-    peakValue = particles['peakMassFlowing']
-    stop = (value <= stopCrit*peakValue) and peakValue > 0
-  elif cfg['stopCritType'] == 'kinEnergy':
-    value = TotkinEneNew
-    peakValue = particles['peakKinEne']
-    stop = value <= stopCrit*peakValue
+
+  if typeStop == 1:
+    totForceSPHNew = totForceSPHNew + mNew * norm(forceSPHX[j], forceSPHY[j], forceSPHZ[j])
```

Comment:
are you sure this line belongs here? I would put it line 737 with the totkinEne and flowing mass. 
Her you only add the sphforce of the last particle no?

Reply:
uh this was moved there not on purpose! It is back to below line 736